scio-bigquery/src/main/scala/com/spotify/scio/bigquery/client/QueryOps.scala

```diff
@@ -253,7 +253,7 @@ private[client] final class QueryOps(client: Client, tableService: TableOps, job
   }
 
   private def isInvalidQuery(e: GoogleJsonResponseException): Boolean =
-    e.getDetails.getErrors.get(0).getReason == "invalidQuery"
+    Try(e.getDetails.getErrors.get(0).getReason).map(_ == "invalidQuery").getOrElse(false)
```

Comment:
Can you do an explicit null check or wrap the call in an `Option` instead of `Try` ?
This code may swallow other exceptions.

Reply:
`isInvalidQuery` is only called when a `Failure` condition has already occurred. If I call in an `Option` the underlying failure will still be suppressed by an NPE (unless I wrap every call in an `Option`).  I will suggest some alternatives even though I still think the solution I originally proposed is the best since what is most important is that the underlying failure is reported as opposed to the exact reason the Google client is returning null.

Reply:
```suggestion
    Option(e.getDetails)
      .map(_.getErrors)
      .filterNot(_.isEmpty)
      .map(_.get(0))
      .exists(_.getReason == "invalidQuery")
```

Reply:
I actually changed my mind.  I like the approach of wrapping in an `Option` despite its verbosity.

Reply:
with a small tweak to your suggestion we should be good.

```suggestion
    Option(e.getDetails)
      .flatMap(details => Option(details.getErrors))
      .flatMap(_.headOption)
      .exists(_.getReason == "invalidQuery")
```

Reply:
Thanks @regadas. I'm assuming you intended `Option(details.getErrors.asScala)`?

Reply:
ahh yes! however it needs to be in the `.flatMap(_.asScala.headOption)`

Reply:
@regadas  I will change it to that, but may I ask why `.flatMap(_.asScala.headOption)` as opposed to `Option(details.getErrors.asScala)`.  It seems to me to be the same result but would interested in learning why the former is preferred.

Reply:
yeah no worries! the first suggestion only works under `2.12. With `2.11` you would still get NPE if `getErrors` returned `null`.

Reply:
That makes sense! Thanks.  The change has been pushed.